src/cli.ts

```diff
@@ -31,17 +31,28 @@ export type CliOptions = z.infer<typeof cliOptions>;
  * @returns the provided options
  */
 export function parseCliArgs() {
-	return argumentParser({
-		options: cliOptions,
-		aliases: {
-			h: 'help',
-			v: 'version',
-			s: 'skipBuild',
-			e: 'experimentalMinify',
-			w: 'watch',
-			c: 'noColor',
-		},
-	}).parse(process.argv.slice(2));
+	try {
+		return argumentParser({
+			options: cliOptions,
+			aliases: {
+				h: 'help',
+				v: 'version',
+				s: 'skipBuild',
+				e: 'experimentalMinify',
+				w: 'watch',
+				c: 'noColor',
+			},
+		}).parse(process.argv.slice(2));
+	} catch (error) {
+		if (error.issues?.[0].code === 'unrecognized_keys') {
+			const unknownKeys = error.issues[0].keys.join(', ');
+			cliError(`Unknown option(s): ${unknownKeys}`);
+			printCliHelpMessage();
+		} else {
+			cliError(error.message);
```

Comment:
Just to be extra sure can we add a fallback message in case the error is not of the correct format?
```suggestion
			cliError(error?.message ??
				'Error: Could not parse the provided Cli arguments.', {spaced: true, showReport: true });
```